tests/test_parse_requirements.py

```diff
@@ -205,3 +244,24 @@ def test_parse_requirements_with_environment_markers(monkeypatch):
     # have chosen foo==3.2.1
     assert set(result) == {"foo"}
     assert str(result["foo"]) == 'foo==3.2.1; python_version > "2.7"'
+
+
+def test_parse_requirements_with_invalid_wheel_filename(monkeypatch):
+    files = {
+        "a.txt": ["https://github.com/pypa/pip/archive/pip-1.3.1-invalid-format.whl"]
+    }
+    monkeypatch.setattr(pip_api._parse_requirements, "_read_file", files.get)
+
+    with pytest.raises(PipError):
+        pip_api.parse_requirements("a.txt")
```

Comment:
Let's assert on exactly what the `PipError` is here

Reply:
9406810